packages/lambda-tiler/src/routes/health.ts

```diff
@@ -40,6 +61,121 @@ export async function updateExpectedTile(test: TestTile, newTileData: Buffer, di
   await fs.promises.writeFile(`${expectedFileName}.diff.png`, imgPng);
 }
 
+async function validateRasterTile(
+  tileSet: ConfigTileSetRaster,
+  test: TestTile,
+  req: LambdaHttpRequest,
+): Promise<LambdaHttpResponse | undefined> {
+  // Get the parse response tile to raw buffer
+  const response = await TileXyzRaster.tile(req, tileSet, test);
+  if (response.status !== 200) return new LambdaHttpResponse(500, response.statusDescription);
+  if (!Buffer.isBuffer(response._body)) throw new LambdaHttpResponse(500, 'Not a Buffer response content.');
+  const resImgBuffer = await Sharp(response._body).raw().toBuffer();
+
+  // Get test tile to compare
+  const testBuffer = await getTestBuffer(test);
+  test.buf = testBuffer;
+  const testImgBuffer = await Sharp(testBuffer).raw().toBuffer();
+
+  const outputBuffer = Buffer.alloc(testImgBuffer.length);
+  const missMatchedPixels = PixelMatch(testImgBuffer, resImgBuffer, outputBuffer, TileSize, TileSize);
+  if (missMatchedPixels) {
+    /** Uncomment this to overwite the expected files */
+    // await updateExpectedTile(test, response._body as Buffer, outputBuffer);
+    req.log.error({ missMatchedPixels, projection: test.tileMatrix.identifier, xyz: test.tile }, 'Health:MissMatch');
+    return new LambdaHttpResponse(500, 'TileSet does not match.');
+  }
+  return;
+}
+
+function propertyCheck(layer: VectorTileLayer, key: string, value: string): boolean {
```

Comment:
could we make this a a `assertLayer` that throws, it would make it easier than all the `if`s

for example

```
assertLayer(tile, 'aeroway', 'name', 'Wellington Airport')
assertLayer(tile, 'place', 'name', 'Wellington')
assertLayer(tile, 'coastline', 'class', 'coastline')
```

Reply:
These checks could even be added into the `TestTile` as a `TestTileVector` eg 

```typescript
  {
    tileSet: 'topographic',
    tileMatrix: GoogleTms,
    tileType: 'pbf',
    tile: { x: 62, y: 40, z: 6 },
    features: [
      { layer: 'aeroway', property: 'name', value: 'Wellington Airport' },
      { layer: 'place', property: 'name', value: 'Wellington' },
      ....
    ]
  },
```